### PR TITLE
[Tines] New integration

### PR DESCRIPTION
**Pull request by colin-stubbs**
Add initial tines integration release

## What does this PR do?

PR addressing "Add Tines integration" [issue #5075](https://github.com/elastic/integrations/issues/5075)

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [X] Linting/Testing/etc

## How to test this PR locally

Signup for a free Tines community account trial and use this integration to pull audit logs from the tenant via API.

## Related issues

- Closes [issue #5075](https://github.com/elastic/integrations/issues/5075)

## Screenshots

Configuration

Dashboard

Dashboard

### CODE REVIEW

File: packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - set:
+      target: header.x-user-email
+      value: {{auth_email}}
+  - set:
+      target: header.x-user-token
+      value: {{auth_token}}
+  - append:
+      target: url.params.after
+      value: '[[.cursor.after]]'
+      default: '[[ formatDate (now (parseDuration "{{initial_lookback}}")) "RFC3339" ]]'
```
Comment (efd6) [inline packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs:24]:
```suggestion
      default: '[[ formatDate (now (parseDuration "-{{initial_interval}}")) "RFC3339" ]]'
```
for consistency with other packages.

File: packages/tines/data_stream/audit_logs/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        required: false
+        show_user: false
+      - name: initial_lookback
+        type: text
+        title: Initial Look Back Period
+        description: Period of historical audit logs to collect when first collection request occurs
+        multi: false
+        required: true
+        show_user: false
+        default: -168h
```
Comment (efd6) [inline packages/tines/data_stream/audit_logs/manifest.yml:67]:
```suggestion
      - name: initial_interval
        type: text
        title: Initial Interval
        description: Period of historical audit logs to collect when first collection request occurs
        multi: false
        required: true
        show_user: false
        default: 168h
```
for consistency with other packages

File: packages/tines/data_stream/audit_logs/sample_event.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (efd6) [inline packages/tines/data_stream/audit_logs/sample_event.json:1]: This appears to have been obtained outside of `elastic-package` (there is no system test that would have generated it and its fields are not lexically ordered). Can you sort the fields or preferably include a system test?

**Review by efd6 — COMMENTED** [review]
This looks good. Are you able to add a system test?

File: packages/tines/data_stream/audit_logs/sample_event.json (hunk 3 above)
Comment (colin-stubbs) [reply packages/tines/data_stream/audit_logs/sample_event.json:1]: Ah yeap, I understand how this works now (finally!), I've done another large commit with full system tests now.

**Comment by jamiehynds** [comment]
@efd6 is there much work left to get this one over the line? We have a webinar coming up with Tines and would be great to showcase @colin-stubbs integration as part of the webinar.

File: packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  - set:
+      target: header.x-user-email
+      value: {{auth_email}}
+  - set:
+      target: header.x-user-token
+      value: {{auth_token}}
+  - set:
+      target: url.params.after
+      value: '[[.cursor.after]]'
+      default: '[[ formatDate (now (parseDuration "{{initial_interval}}")) "RFC3339" ]]'
```
Comment (efd6) [inline packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs:24]:
```suggestion
      default: '[[ formatDate (now (parseDuration "-{{initial_interval}}")) "RFC3339" ]]'
```

File: packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+{{/if}}
+{{#if operation_name}}
+  - set:
+      target: url.params.operation_name
+      value: "{{operation_name}}"
+{{/if}}
+cursor:
+  after:
+    value: '[[.first_event.updated_at]]'
+    default: '[[ formatDate (now (parseDuration "{{initial_lookback}}")) "RFC3339" ]]'
```
Comment (efd6) [inline packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs:41]:
```suggestion
    default: '[[ formatDate (now (parseDuration "-{{initial_lookback}}")) "RFC3339" ]]'
```

File: packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  - set:
+      target: header.x-user-email
+      value: {{auth_email}}
+  - set:
+      target: header.x-user-token
+      value: {{auth_token}}
+  - set:
+      target: url.params.start_date
+      value: '[[.cursor.start_date]]'
+      default: '[[ formatDate (now (parseDuration "{{initial_interval}}")) "RFC3339" ]]'
```
Comment (efd6) [inline packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs:24]:
```suggestion
      default: '[[ formatDate (now (parseDuration "-{{initial_interval}}")) "RFC3339" ]]'
```

File: packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+{{/if}}
+{{#if story_id}}
+  - set:
+      target: url.params.story_id
+      value: "{{story_id}}"
+{{/if}}
+cursor:
+  start_date:
+    value: '[[.last_event.time_saved.date]]'
+    default: '[[ formatDate (now (parseDuration "{{initial_lookback}}")) "RFC3339" ]]'
```
Comment (efd6) [inline packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs:51]:
```suggestion
    default: '[[ formatDate (now (parseDuration "-{{initial_lookback}}")) "RFC3339" ]]'
```

File: packages/tines/data_stream/time_saved/manifest.yml [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        required: false
+        show_user: false
+      - name: initial_interval
+        type: text
+        title: Initial Interval
+        description: Period of historical audit logs to collect when first collection request occurs
+        multi: false
+        required: true
+        show_user: false
+        default: -168h
```
Comment (efd6) [inline packages/tines/data_stream/time_saved/manifest.yml:83]:
```suggestion
        default: 168h
```

File: packages/tines/data_stream/audit_logs/manifest.yml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        required: false
+        show_user: false
+      - name: initial_interval
+        type: text
+        title: Initial Interval
+        description: Period of historical audit logs to collect when first collection request occurs
+        multi: false
+        required: true
+        show_user: false
+        default: -168h
```
Comment (efd6) [inline packages/tines/data_stream/audit_logs/manifest.yml:67]:
```suggestion
        default: 168h
```

**Review by efd6 — COMMENTED** [review]
Apologies for the delay. Minor changes only for consistency with other packages.

**Comment by efd6** [comment]
There appears to be a conflict in CODEOWNERS
```
/packages/ti_recordedfuture @elastic/security-external-integrations
/packages/ti_threatq @elastic/security-external-integrations
/packages/ti_util @elastic/security-external-integrations
<<<<<<< HEAD
/packages/tomcat @elastic/obs-service-integrations
=======
/packages/tines @elastic/security-external-integrations
/packages/tomcat @elastic/security-external-integrations
>>>>>>> 79b49a153 (Initial commit for tines integration)
/packages/traefik @elastic/obs-service-integrations
/packages/trend_micro_vision_one @elastic/security-external-integrations
/packages/udp @elastic/security-external-integrations
```

@colin-stubbs I'm happy to resolve this if you are OK with that.

**Comment by colin-stubbs** [comment]
> There appears to be a conflict in CODEOWNERS
> 
> ```
> /packages/ti_recordedfuture @elastic/security-external-integrations
> /packages/ti_threatq @elastic/security-external-integrations
> /packages/ti_util @elastic/security-external-integrations
> <<<<<<< HEAD
> /packages/tomcat @elastic/obs-service-integrations
> =======
> /packages/tines @elastic/security-external-integrations
> /packages/tomcat @elastic/security-external-integrations
> >>>>>>> 79b49a153 (Initial commit for tines integration)
> /packages/traefik @elastic/obs-service-integrations
> /packages/trend_micro_vision_one @elastic/security-external-integrations
> /packages/udp @elastic/security-external-integrations
> ```
> 
> @colin-stubbs I'm happy to resolve this if you are OK with that.

No issues from my side!

File: packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs (hunk 4 above)
Comment (colin-stubbs) [reply packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs]: Resolved via next commit.

File: packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs (hunk 5 above)
Comment (colin-stubbs) [reply packages/tines/data_stream/audit_logs/agent/stream/httpjson.yml.hbs]: Resolved via next commit. Also fixed name of variable.

File: packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs (hunk 6 above)
Comment (colin-stubbs) [reply packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs]: Resolved via next commit.

File: packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs (hunk 7 above)
Comment (colin-stubbs) [reply packages/tines/data_stream/time_saved/agent/stream/httpjson.yml.hbs]: Resolved via next commit. Also fixed name of variable.

File: packages/tines/data_stream/audit_logs/manifest.yml (hunk 9 above)
Comment (colin-stubbs) [reply packages/tines/data_stream/audit_logs/manifest.yml]: Resolved via next commit.

**Comment by colin-stubbs** [comment]
@efd6 I tried to fix the conflict in CODEOWNERS on that last commit, but I'm now not entirely sure what's complaining about...

**Comment by v1v** [comment]
@mergifyio rebase

**Comment by v1v** [comment]
@Mergifyio update

**Comment by efd6** [comment]
@colin-stubbs Can you enable maintainer changes on this PR?

**Comment by colin-stubbs** [comment]
> @colin-stubbs Can you [enable maintainer changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) on this PR?

Yeah I don't see that option at all... perhaps because it's under an org/team? Either way, I've added you with write access to my fork now, which should be enough?

**Comment by efd6** [comment]
@colin-stubbs Thanks. After we've got this all happy, I'll drop that access.

**Review by efd6 — APPROVED** [review]
Thanks

**Comment by jamiehynds** [comment]
@efd6 I just tried to find the Tines package in the integrations page and the EPR, but no sign of it. Any ideas? I noticed the package version is set to 0.0.1, should it be 0.1.0 in order to appear in the EPR?

**Comment by efd6** [comment]
It's prelease, https://epr.elastic.co/search?package=tines&prerelease=true (https://github.com/elastic/package-registry#search, thanks @andrewkroh).

**Comment by jamiehynds** [comment]
Just realised the minimum Kibana version is [set to](https://github.com/elastic/integrations/blob/main/packages/tines/manifest.yml#L11) 8.6.1 - is it possible to adjust to 8.6.x, assuming there's no dependency on 8.6.1 specifically?

**Comment by colin-stubbs** [comment]
None that I'm aware of. It's only set to that as I've developed on the latest version of elastic-package and that was running a stack using 8.6.1

**Pull request merged**